master/security/domain-based-policy.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Admin policy using domain names
+canonical_url: 'https://docs.projectcalico.org/master/security/domain-based-policy'
+---
+
+In a Kubernetes deployment, the cluster administrator can configure
+[GlobalNetworkPolicy]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/globalnetworkpolicy)
+to allow egress traffic from the cluster to peers that are identified
+by their domain names.  {{site.prodname}} has long supported
+identifying peers directly by IP, but it will usually be more
+meaningful and resilient to identify a resource *outside* the
+immediate Kubernetes cluster - such as a cloud-provider database like
+AWS's RDS - by its domain name instead of its IP.
+
+> **Note**: Similarly, in order to address endpoints and services
+> *within* the cluster, we apply Kubernetes labels to convey their
+> roles, and then select on those labels in Calico policy, rather than
+> addressing those endpoints and services directly by IP.  So using
+> domain names for external resources is somewhat analogous to using
+> labels for internal ones.  Using domain names for services *within*
+> the cluster is not supported by the feature described on this page,
+> because Kubernetes labels can and should be used for those instead.
+{: .alert .alert-info}
+
+Specifically, a GlobalNetworkPolicy can have egress rules with
+`action: Allow` and a `domains` field specifying the possible domain
+names to which egress traffic is allowed.
+
+> **Note**: This makes sense when egress traffic from the workloads
+> concerned (i.e. that the policy applies to) is denied by default.
+> In other words, so that the GlobalNetworkPolicy with domain names is
+> poking holes in a generally closed wall around the cluster.
+{: .alert .alert-info}
+
+There are two ways for a rule to specify allowed domain names:
+
+1.  By listing those domain names directly in `domains.names`.
+
+2.  By creating a
+    [GlobalDomainSet]({{site.baseurl}}/{{page.version}}/reference/calicoctl/resources/globaldomainset)
+    resource - which groups together a set of domain names that you
+    want to label in the same way - and using `domains.selector` to
+    select GlobalDomainSets by their labels.  The effect then is to
+    allow egress to all of the domain names listed in all of the
+    selected GlobalDomainSets.
+
+### Understanding which domain names will work
```

Comment:
I would like to review this section after we have got answers to the requirements questions from PM and after we've agreed the design we're going to implement.

Reply:
More precisely, I agree with reviewing after we're sure about requirements.  I think doc/spec/UX review should be before or at the same time as design review, not _after_ design review, surely?

Reply:
Yes, at the same time as design is more precise.  Agree!